test/fork/withdraw-enzyme-shares.js

```diff
@@ -0,0 +1,146 @@
+        console.log(`Reverted to snapshot ${TENDERLY_SNAPSHOT_ID}`);
+      } else {
+        console.log('Snapshot ID: ', await evm.snapshot());
+      }
+    }
+  });
+
+  it('load contracts', async function () {
+    this.master = await ethers.getContractAt('NXMaster', '0x01BFd82675DBCc7762C84019cA518e701C0cD07e');
+    this.swapOperator = await ethers.getContractAt('SwapOperator', V2Addresses.SwapOperator);
+    this.nxmToken = await ethers.getContractAt('NXMToken', NXM_TOKEN_ADDRESS);
+
+    this.pool = await ethers.getContractAt('Pool', await this.master.getLatestAddress(toUtf8Bytes('P1')));
+
+    this.enzymeVaultShares = await ethers.getContractAt('ERC20Mock', ENZYMEV4_VAULT_PROXY_ADDRESS);
+
+
+    const governanceAddress = await this.master.getLatestAddress(toUtf8Bytes('GV'));
+
+    await evm.impersonate(governanceAddress);
+    await evm.setBalance(governanceAddress, parseEther('1000'));
+    this.governanceImpersonated = await getSigner(governanceAddress);
+  });
+
+  it.skip('Impersonate AB members', async function () {
+    const { memberArray: abMembers } = await this.memberRoles.members(1);
+    this.abMembers = [];
+    for (const address of abMembers) {
+      await evm.impersonate(address);
+      await evm.setBalance(address, parseEther('1000'));
+      this.abMembers.push(await getSigner(address));
+    }
+  });
+
+
+  it('Upgrade SwapOperator', async function () {
+
+    const swapControllerAddress = await this.swapOperator.swapController();
+
+
+    const newSwapOperator = await ethers.deployContract('SwapOperator', [
+        COWSWAP_SETTLEMENT,
+        swapControllerAddress,
+        this.master.address,
+        WETH_ADDRESS,
+        ENZYMEV4_VAULT_PROXY_ADDRESS,
+        ENZYME_FUND_VALUE_CALCULATOR_ROUTER,
+        '0'
+    ]);
+
+    await this.pool.connect(this.governanceImpersonated).updateAddressParameters(
+      toBytes8('SWP_OP'),
+      newSwapOperator.address
+    );
+
+    this.swapOperator = await ethers.getContractAt('SwapOperator', newSwapOperator.address);
+  });
+
+  it('withdraw enzyme', async function () {
+
+    const swapControllerAddress = await this.swapOperator.swapController();
+    await evm.impersonate(swapControllerAddress);
+    await evm.setBalance(swapControllerAddress, parseEther('1000'));
+    const swapController = await getSigner(swapControllerAddress);
+
+    const shareAmount = await this.enzymeVaultShares.balanceOf(this.pool.address);
+
+    console.log({
+      shareAmount: shareAmount.toString()
+    });
+
+    await this.pool.connect(this.governanceImpersonated).setSwapDetails(
+      ENZYMEV4_VAULT_PROXY_ADDRESS,
+      '0',
+      '1',
+      '10000' // max slippage ratio
+    );
+
+    await enableAsEnzymeReceiver(this.swapOperator.address);
+
+    const swapAmount = shareAmount; //  parseEther('100');
+    const minOut = shareAmount.mul(84).div(100);
+    await this.swapOperator.connect(swapController).swapEnzymeVaultShareForETH(swapAmount, minOut);
```

Comment:
good news this works as soon as we fix the bug.

You get a ratio of the ETH. 85% or so.

Reply:
you have to use the oracle here instead of this hardcoded value. this is basically the exchange rate

Reply:
Tried it and added code that uses that but enzyme returns something under the oracle price (in the 20-25 ether range less).